web/app.js

```diff
@@ -683,6 +686,43 @@ const PDFViewerApplication = {
     } else {
       throw new Error("Not implemented: run");
     }
+
+    if (
+      (typeof PDFJSDev === "undefined" || !PDFJSDev.test("MOZCENTRAL")) &&
+      !("onscrollend" in document.documentElement)
+    ) {
+      return;
+    }
+
+    const { mainContainer } = appConfig;
+    ({ scrollTop: this._lastScrollTop, scrollLeft: this._lastScrollLeft } =
+      mainContainer);
+    const scroll = () => {
+      if (
+        this._lastScrollTop === mainContainer.scrollTop &&
+        this._lastScrollLeft === mainContainer.scrollLeft
+      ) {
+        return;
+      }
+      mainContainer.removeEventListener("scroll", scroll, {
+        passive: true,
+      });
+      this._isScrolling = true;
+      const scrollend = () => {
+        this._lastScrollTop = mainContainer.scrollTop;
```

Comment:
Don't we need to update the left property as well, since scrolling could happen *only horizontally* as well?